api/src/main/java/io/strimzi/api/kafka/model/connect/build/DownloadableArtifact.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.connect.build;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.strimzi.api.kafka.model.Constants;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.strimzi.crdgenerator.annotations.Pattern;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+/**
+ * Artifact which can be just downloaded from an URL
+ */
+@Buildable(
+        editableEnabled = false,
+        builderPackage = Constants.FABRIC8_KUBERNETES_API
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@EqualsAndHashCode
+public abstract class DownloadableArtifact extends Artifact {
+    private static final long serialVersionUID = 1L;
+
+    private String url;
+    private String sha512sum;
+
+    @Description("URL of the artifact which will be downloaded. " +
+            "Strimzi does not do any security scanning of the downloaded artifacts. " +
+            "For security reasons, you should first verify the artifacts manually and configure the checksum verification to make sure the same artifact is used in the automated build. " +
+            "Required.")
+    @JsonProperty(required = true)
+    @Pattern("^(https?|ftp)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]")
```

Comment:
Did you get this from somewhere, or construct it yourself?

Reply:
I'm quite sure I got it from somewhere. But I do not remember where from anymore TBH. Probably Stack Overflow :-o.